### ISSUE TITLE
videos display

### ISSUE THREAD

**Issue by yael-ai** [eleurent/highway-env#217]
Hey
The output videos that I am getting are extremely fast..
For a duration of 400 episodes, simulation frequency 15 and policy frequency 1 I am getting video's of 5 seconds.
I wish to get longer and slower videos is it possible?

Thanks in advance

**Comment by eleurent** [comment]
Hi,

Yes, that is because openai gym is configured such that a video frame is taken at every `env.step()`, so whenever an action is executed, that is, at 1 Hz with your policy frequency. But the video fps is set to a much higher value, like 24, which explains why it plays so fast. 

Here are a few possibilities to get better videos:

- Keep policy frequency at 1Hz, but set the video fps to 1Hz as well. This way, you will get "real-time" videos, and an episode of 30 steps should generate a video lasting 30s. However, the video will play at 1 fps which is jerky and terrible.
- Increase the policy frequency to a higher value, like 15Hz, and set the video fps to the same value. You will also get "real-time" videos, but with much more frames in it (higher fps), which will look smoother and better. However, this will also affect the training of the RL agent (the discount factor, buffer size, etc must be changed accordingly).
- You can also keep a policy frequency of 1 Hz, but use a little hack to record every simulated frame (at 15Hz) rather than only the last generated frame for this action. I implemented it here:
https://github.com/eleurent/highway-env/blob/d6ef0d6d766c3b54a9f06cba3f09677d718109c3/scripts/utils.py#L27

Basically, it we set the `Monitor.video_recorder.capture_frame` method as a callback which is called whenever a simulation step is performed, so that these frames are also added to the video. See https://colab.research.google.com/github/eleurent/highway-env/blob/master/scripts/highway_planning.ipynb for an example of how it is used.

**Comment by eleurent** [comment]
See https://highway-env.readthedocs.io/en/latest/faq.html
